### ISSUE TITLE
Reliability issues (pictures permanently lost)

### ISSUE THREAD

**Issue by mnalis** [nobelization/panoramax-mobile-app#132]
I'm having many reliability issues (with 1.2.0-beta). 

App often freezes, especially if anything at all interrupts it (for example, alarm going off, incoming phone call, switching to other app, etc.). Usual symptom is that the app stops taking pictures and freezes on last taken picture, and/or that it gets stuck on that "wavy form" animation (that usually only takes a fraction of the second between taking pictures).

When that happens, anything I try is in vain, and ***whole recordings with hundres/thousands of pictures is lost***.
And it happens quite often for me (Android 14) - I'd estimate about 70% of all of  my panoramax  app recordings attempts were completely lost because app was interrupted while taking pictures, with about 10% extra were partially lost while being interrupted during upload (usually low quality mobile data or other app interruption - i.e. https://github.com/nobelization/panoramax-mobile-app/issues/119). Thus, I've only been able to contribute short segments, when most of the long ones were all lost.

I've tried to exempt the app from any Android power saving in attempt to address at least those issues perhaps caused by https://github.com/nobelization/panoramax-mobile-app/issues/100, but it did not really help.

Here is one example:

https://github.com/user-attachments/assets/e25ba5c8-c59d-4559-a4e2-6342170ac45c

---

Suggestions:

- Would it be possible to somehow preserve the pictures / track in case of an errors (i.e. allow storing them in publicly accessible folder on non-rooted phones)? So one can at least attempt to manually take the files off the phone and later attempt to upload them manually via CLI tools? 

- of course, generally improving the reliability of the app would be great, to make it able to resume taking pictures after interruptions, breaking out of such "Hangs"  and continuing with picture taking (or at least uploading that part of the track) etc.

- Also, I would like to suggest holding off with making non-beta releases to wider audiences (i.e. [f-droid](https://github.com/nobelization/panoramax-mobile-app/pull/38), Google play etc) until those reliability issues are addressed, as people who try the app to record the pictures and have their effort of collecting pictures completely wasted would likely be very much put down, and give up on even trying newer versions of the app again.

**Comment by chrisdebian** [comment]
Yes, I know what you mean. When the wavy animation won't go away, all I can do is reboot.

Chris

**Comment by overflorian** [comment]
I share your concern and frustration because I face the same issues.
> Would it be possible to somehow preserve the pictures / track in case of an errors (i.e. allow storing them in publicly accessible folder on non-rooted phones)

Yes, that was already on the roadmap but it's higher in the priority list now ;)

> generally improving the reliability of the app

I agree too. We'll see how to do that.

> When the wavy animation won't go away, all I can do is reboot.

We have planned to make the STOP button clickable even when the wavy animation is playing. I believe that is a first step in the right direction.

> Also, I would like to suggest holding off with making non-beta releases to wider audiences (i.e. https://github.com/nobelization/panoramax-mobile-app/pull/38, Google play etc) until those reliability issues are addressed

We really would like to publish the app this year so we decided to publish it even if that's not fixed. Even for testing, it's really easier to have a proper delivery channel with integrated updates.  
But we are also conscious of the need for improvement, so we won't make any public publicity before reaching a _stable enough_ version.

**Comment by overflorian** [comment]
>  But we are also conscious of the need for improvement, so we won't make any public publicity before reaching a stable enough version.

Also, after much consideration of your proposal, the publication will be limited to F-Droid for now.

**Comment by BDVGitHub** [comment]
@overflorian So if I understand it correctly after the capture off a photo or sequence we need a screen where you have the option to send the photo's or store the photo's?

**Comment by BDVGitHub** [comment]
@OtterWays or @overflorian Is it possible to get me the latest version of screens to capture images? Can't find these directly in Penpot. So that I can customise them by adding a button to send the images later.

@OtterWays I have exported the Penpot Panoramax library as (.svg + json) could you see if you can open these files and use them to implement later in the UI/UX? thanks.
[Panoramax Mobile App Library.zip](https://github.com/user-attachments/files/17988448/Panoramax.Mobile.App.Library.zip)

The background would look like this, this one below is with the top banner with the back button, there is also a top banner with hamburger (menu) icon.
![afbeelding](https://github.com/user-attachments/assets/5132b6db-c427-4b88-8eae-e67170f58328)
